tuf_conformance/test_updater_key_rotations.py

```diff
@@ -145,3 +173,61 @@ def test_root_rotation(client: ClientRunner,
 
     with open(os.path.join(client.metadata_dir, "root.json"), "rb") as f:
         assert f.read() == expected_local_root
+
+
+@pytest.mark.parametrize("md_version", non_rotation_cases, ids=non_rotation_ids)
+def test_non_root_rotations(client: ClientRunner,
+                            server: SimulatorServer,
+                            md_version: MdVersion) -> None:
+    """Test Updater.refresh() with various sequences of metadata updates
+
+    Each MdVersion in the list describes metadata keys and signatures
+    of a remote metadata version. As an example:
+        MdVersion([1,2,3], 2, [1,2])
+    defines a metadata that contains keys 1, 2 and 3 with threshold 2. The
+    metadata is signed with keys 1 and 2.
+
+    Assert that refresh() result is expected and that local metadata on disk
+    is the expected one after all roots have been loaded from remote using
+    the standard client update workflow.
+    """
+    signers = []
+    for _ in range(10):
+        signer = CryptoSigner.generate_ed25519()
+        signers.append(signer)
+
+    # initialize a simulator with repository content we need
+    init_data, repo = server.new_test(client.test_name)
+    assert client.init_client(init_data) == 0
+
+    roles = ["timestamp", "snapshot", "targets"]
+    for role in roles:
+        # clear role keys, signers
+        repo.root.roles[role].keyids.clear()
+        repo.signers[role].clear()
+
+        repo.root.roles[role].threshold = md_version.threshold
+        for i in md_version.keys:
+            repo.root.add_key(signers[i].public_key, role)
+
+        for i in md_version.sigs:
+            repo.add_signer(role, signers[i])
+
+        repo.root.version += 1
+        repo.publish_root()
+
+        # run client workflow, assert success/failure
+        expected_error = md_version.res
+        if expected_error:
```

Comment:
naming is not great: when the test is expecting success, `expected_error==True`

Reply:
https://github.com/theupdateframework/tuf-conformance/pull/104/commits/bdc11da0cac16015bcca384c369e94c58ee36ef4